pkg/kubecost/allocation.go

```diff
@@ -2431,12 +2494,26 @@ func (asr *AllocationSetRange) UTCOffset() time.Duration {
 // Window returns the full window that the AllocationSetRange spans, from the
 // start of the first AllocationSet to the end of the last one.
 func (asr *AllocationSetRange) Window() Window {
-	if asr == nil || asr.Length() == 0 {
+	asr.Lock()
+	defer asr.Unlock()
+	return asr.window()
+}
+
+func (asr *AllocationSetRange) window() Window {
```

Comment:
I'd like to see a comment here explaining that `.window()` isn't safe for concurrent access unless the lock has already been acquired.